### ISSUE TITLE
[Bug] `remark-lint-no-undefined-references` false positive on `.github/ISSUE_TEMPLATE/*.md` with square brackets (such as `title: "[Improve]"`).

### ISSUE THREAD

**Issue by SwuduSusuwu** [remarkjs/remark-lint#324]
### Initial checklist

- [x] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [x] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [x] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [x] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=issues) and [discussions](https://github.com/orgs/remarkjs/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

_GitHub_'s workflow (https://github.com/SwuduSusuwu/SubStack/blob/SwuduSusuwu-issues-improve/.github/workflows/codacy.yml)

### Steps to reproduce

- Enable `Remark-lint` _GitHub_ workflows (such as https://github.com/codacy + https://github.com/rtfpessoa/codacy-remark-lint)
  - https://github.com/SwuduSusuwu/SubStack/blob/SwuduSusuwu-issues-improve/.github/workflows/codacy.yml
- Use https://github.com/SwuduSusuwu/SubStack/issues/templates/edit to add a new template (\<template\>) for issues, with square brackets for the default title (`title: "[Improve] <relative path>: <what to do>"`)
  - This commits to https://github.com/\<user\>/\<repo\>/tree/\<user\>-patch/.github/ISSUE_TEMPLATE/\<template\>.md

### Actual behavior

https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2332
> .github/ISSUE_TEMPLATE/feature_request.md:4
> ---
> name: Feature request
> about: Suggest an idea for this project
> title: "[Improve] \<relative path\>: \<what to do\>"
>> [no-undefined-references] Found reference to undefined definition
Remark-lint (reported by Codacy)

### Expected behavior

https://github.com/SwuduSusuwu/SubStack/pull/51/commits/719afd57da2b74f191513b56ed3d1fa54fed51c3 passes

### Runtime

GitHub workflows

### Package manager

GitHub workflows

### Operating system

GitHub workflows

**Comment by SwuduSusuwu** [comment]
If you [escape those](https://github.com/SwuduSusuwu/SubStack/commit/9d04328c1658d2a38163631a7209afa44e37b9d5), [_GitHub_ fails](https://github.com/SwuduSusuwu/SubStack/blob/9d04328c1658d2a38163631a7209afa44e37b9d5/.github/ISSUE_TEMPLATE/feature_request.md) with
> Error in user YAML: (<unknown>): found unknown escape character while parsing a quoted scalar at line 3 column 8
>
> ---
> name: Feature request
> about: Suggest how to improve code/workflow
> title: "\[Improve\] <relative path>: <what to do>"

**Comment by ChristianMurphy** [comment]
@SwuduSusuwu it sounds like you are using front matters but have not added/enabled `remark-frontmatter`.

**Comment by SwuduSusuwu** [comment]
> [@SwuduSusuwu](https://github.com/SwuduSusuwu) it sounds like you are using front matters but have not added/enabled `remark-frontmatter`.

https://github.com/remarkjs/remark-frontmatter sounds like a solution, but don't know how to include that into _GitHub_'s workflows (or if you can).

Have not found selective measures (such as `clang-tidy`'s `/* NOLINT(concurrency-mt-unsafe): is unreachable code */` or `shellcheck`'s `#shellcheck disable=SC2086`) to disable this. Hope someone knows how to

**Comment by ChristianMurphy** [comment]
> sounds like a solution, but don't know how to include that into GitHub's workflows (or if you can).

I'm not sure which GitHub workflow you are referring to.
And no I don't manage any of them.
This would be something to open an issue with that tool to have them add the frontmatter plugin.

There are disables supported by remark https://github.com/remarkjs/remark-message-control
But those may cause conflicts with whatever other tool isn't correctly configuring remark in the first place.

**Comment by SwuduSusuwu** [comment]
> There are disables supported by remark https://github.com/remarkjs/remark-message-control But those may cause conflicts with whatever other tool isn't correctly configuring remark in the first place.

https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2332
> [.github/ISSUE_TEMPLATE/feature_request.md:5](https://github.com/SwuduSusuwu/SubStack/blob/8cb8feee49383c1ef353ab47f7b052d38470a1cd/.github/ISSUE_TEMPLATE/feature_request.md#L5-L5)
> name: Feature request
> about: Suggest how to improve code/workflow
> 
> title: "[Improve] <relative path>: <what to do>"
>> [no-undefined-references] Found reference to undefined definition
>> Remark-lint (reported by Codacy)

Want to know if this is wrong use of `lint ignore` or if _GitHub_'s workflows just don't support this.

**Comment by SwuduSusuwu** [comment]
https://github.com/codacy/codacy-remark-lint/blob/master/package.json#L68=L75 shows `"dependencies": {` ... `"remark-frontmatter": "^3.0.0",`; know that is for `frontmatter` use.
- [https://github.com/\<user\>/\<repo\>/blob/\<branch\>/.github/workflows/codacy.yml](https://github.com/SwuduSusuwu/SubStack/blob/SwuduSusuwu-issues-improve/.github/workflows/codacy.yml) does not seem to have an option for this.

**Comment by SwuduSusuwu** [comment]
"Enable `remark-frontmatter`" is not a solution;
- The _GitHub_ tools do use `remark-frontmatter`, which is the reason that the error is not triggered on all `.github/ISSUE_TEMPLATES/*.md` (just on those which use brackets).
- The [error which occurs if `remark-frontmatter` is not enabled](https://answers.netlify.com/t/getting-issue-on-codacy-because-of-3-dashes/2762) is different from this `[no-undefined-references] Found reference to undefined definition`

**Comment by ChristianMurphy** [comment]
> The [error which occurs if remark-frontmatter is not enabled](https://answers.netlify.com/t/getting-issue-on-codacy-because-of-3-dashes/2762) is different from this [no-undefined-references] Found reference to undefined definition

Not enabling frontmatter is the root of the problem.
You need it enabled.

**Comment by SwuduSusuwu** [comment]
> > The [error which occurs if remark-frontmatter is not enabled](https://answers.netlify.com/t/getting-issue-on-codacy-because-of-3-dashes/2762) is different from this [no-undefined-references] Found reference to undefined definition
> 
> Not enabling frontmatter is the root of the problem. You need it enabled.

It is, which is the reason that:
```
---
name: Feature request
about: Suggest an idea for this project
title: "Improve <relative path>: <what to do>"
````
(without brackets) does not give errors.
The problem is that

```
---
name: Feature request
about: Suggest an idea for this project
title: "[Improve] <relative path>: <what to do>"
````
(with brackets) gives errors, and the command to suppress those errors (which is `<!--lint ignore remark-lint-no-undefined-references-->`) causes _YAML_ errors.

**Comment by wooorm** [comment]
Please enable `remark-frontmatter`, as mentioned several times now, that is the root of your problem.
